### PR TITLE
CASVolume spec update and Feature Gate check update

### PR DESCRIPTION
**Pull request by princerachit**
**What this PR does / why we need it**:
Introduced new properties in CASVolumeSpec and CStorVolumeSpec. Updated logic to check CAS template gate feature.

### CODE REVIEW

File: pkg/util/env_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	}{
+		"Missing env variable": {
+			key:         "_MY_MISSING_TEST_KEY_",
+			expectValue: "false",
+		},
+		"Present env variable": {
+			key:         "_MY_PRESENT_TEST_KEY_",
+			expectValue: "value1",
+		},
+	}
```
Comment (prateekpandey14) [inline pkg/util/env_test.go]: Add a case where key is present without any value ?

File: pkg/util/env_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	feature, err := CASTemplateFeatureGate()
+	if err != nil {
+		t.Errorf("expected no error when feature gate %s is unset", CASTemplateFeatureGateENVK)
+	}
+	if feature {
+		t.Errorf("expected false when feature gate %s is unset", CASTemplateFeatureGateENVK)
+	}
+
+	// Positive testcase, env variable set to true
+	os.Setenv(string(CASTemplateFeatureGateENVK), "true")
```
Comment (prateekpandey14) [inline pkg/util/env_test.go]: Same unit test case as commented above.

Comment (princerachit) [reply pkg/util/env_test.go]: done

File: pkg/util/env_test.go (hunk 1 above)
Comment (princerachit) [reply pkg/util/env_test.go]: done

File: pkg/util/env_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"testing"
+	"os"
+	"github.com/magiconair/properties/assert"
+)
+
+func TestLookEnv(t *testing.T) {
+	// set required environment
+	os.Setenv("_MY_PRESENT_TEST_KEY_", "value1")
```
Comment (prateekpandey14) [inline pkg/util/env_test.go]: Add `defer os.Unset(_MY_PRESENT_TEST_KEY_)`   ?

File: cmd/maya-apiserver/app/server/volume_endpoint.go [hunk 4]
```diff
@@ -22,7 +22,12 @@ const (
 // requests w.r.t a single Volume.
 func (s *HTTPServer) volumeSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
 	// check the feature gate & switch if enabled
-	if util.CASTemplateFeatureGate() {
+	feature, err := util.CASTemplateFeatureGate()
+	if err != nil {
+		// exit if invalid value
+		glog.Fatalf("invalid feature gate value for %s only boolean values allowed", util.CASTemplateFeatureGateENVK)
```
Comment (AmitKumarDas) [inline cmd/maya-apiserver/app/server/volume_endpoint.go]: We should not panic here !!! We return error i.e. 500 http code

File: cmd/maya-apiserver/app/server/volume_endpoint.go [hunk 5]
```diff
@@ -22,7 +22,13 @@ const (
 // requests w.r.t a single Volume.
 func (s *HTTPServer) volumeSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
 	// check the feature gate & switch if enabled
-	if util.CASTemplateFeatureGate() {
+	feature, err := util.CASTemplateFeatureGate()
+	if err != nil {
+		// log and return http error 500
+		glog.Errorf("invalid feature gate value for %s only boolean values allowed", util.CASTemplateFeatureGateENVK)
```
Comment (avishnu) [inline cmd/maya-apiserver/app/server/volume_endpoint.go]: Log message should start with upper case.

**Pull request merged**